### DISCUSSION TITLE
How to output 10 predicted structures at one time and how to correctly input HEME as ligand

### DISCUSSION THREAD

**Discussion by KunShi-ECUST** [chaidiscovery/chai-lab#175] [Q&A] [answered]
Hello! I would like to output more (say, 10) predicted results at one time in the locally deployed Chai-1 by me. However, I haven't found relevant content in the tutorial. Could you please provide me with the method?
Besides, I'm unable to input the Smiles file of HEME correctly, and it will report errors. So, currently, I split it into an Fe ion and a porphyrin ring. But the conformation of HEME in the output results is not very accurate. I'm looking forward to your help. Thank you!

**Comment by Ruibin-Liu** [comment]
My understanding is to change the value 5 in this line to 10 if you want 10.
https://github.com/chaidiscovery/chai-lab/blob/44375d5d4ea44c0b5b7204519e63f40b063e4a7c/chai_lab/chai1.py#L535

Not sure about the HEME problem.

**Reply by KunShi-ECUST** [reply]
I have already changed it to 10, but it still outputs 5 results.

**Reply by jackdent** [reply]
Updating this constant won't work: the model will always output 5 samples, since that is fixed in the exported model graph.

**Comment by jackdent** [comment, marked as answer]
The simplest way to do this is to run `run_inference` multiple times in a loop. Each iteration will produce 5 samples.

**Comment by sharonJXY** [comment]
Hi! I run into the same HEME issue. Did you finally find a way to solve it?
